File: src/ros2_bridge/mappers/geometry_msgs/wrench_stamped.rs
//! Typed mapper for `geometry_msgs/msg/WrenchStamped`.

use crate::ros2_bridge::mapper::TypedTopicMapper;

pub(crate) fn wrench_stamped_to_bus(msg: ros_env::geometry_msgs::msg::WrenchStamped) -> crate::geometry_msgs::msg::v1::WrenchStamped {
    crate::geometry_msgs::msg::v1::WrenchStamped {
        header: Some(crate::ros2_bridge::mappers::std_msgs::header::header_to_bus(msg.header)),
        wrench: Some(crate::ros2_bridge::mappers::geometry_msgs::wrench::wrench_to_bus(msg.wrench)),
    }
}

pub(crate) fn wrench_stamped_to_ros(bus: crate::geometry_msgs::msg::v1::WrenchStamped) -> ros_env::geometry_msgs::msg::WrenchStamped {
    ros_env::geometry_msgs::msg::WrenchStamped {
        header: crate::ros2_bridge::mappers::std_msgs::header::header_to_ros(bus.header.unwrap_or_default()),
        wrench: crate::ros2_bridge::mappers::geometry_msgs::wrench::wrench_to_ros(bus.wrench.unwrap_or_default()),
    }
}

#[derive(Clone, Copy, Debug, Default)]
pub struct GeometryMsgsWrenchStampedMapper;

impl TypedTopicMapper for GeometryMsgsWrenchStampedMapper {
    type Ros = ros_env::geometry_msgs::msg::WrenchStamped;
    type Bus = crate::geometry_msgs::msg::v1::WrenchStamped;

    fn ros_to_bus(&self, msg: Self::Ros) -> crate::errors::Result<Self::Bus> {
        Ok(wrench_stamped_to_bus(msg))
    }

    fn bus_to_ros(&self, msg: Self::Bus) -> crate::errors::Result<Self::Ros> {
        Ok(wrench_stamped_to_ros(msg))
    }
}

File: bindings/python/robot_bus/ros2_bridge/mappers/geometry_msgs/wrench_stamped.py
"""Generated mapper for `geometry_msgs/msg/WrenchStamped`."""

from __future__ import annotations

from robot_bus.ros2_bridge.mappers import _convert
from robot_bus.ros2_bridge.mappers.std_msgs.header import header_to_bus, header_to_ros
from robot_bus.ros2_bridge.mappers.geometry_msgs.wrench import wrench_to_bus, wrench_to_ros

_BusMsg = None


def _bus_cls():
    global _BusMsg
    if _BusMsg is None:
        from robot_bus.geometry_msgs.msg.v1 import WrenchStamped as BusMsg

        _BusMsg = BusMsg
    return _BusMsg


def wrench_stamped_to_bus(msg):
    BusMsg = _bus_cls()
    bus = BusMsg()
    bus.header.CopyFrom(header_to_bus(msg.header))
    bus.wrench.CopyFrom(wrench_to_bus(msg.wrench))
    return bus


def wrench_stamped_to_ros(bus):
    from geometry_msgs.msg import WrenchStamped as RosMsg

    out = RosMsg()
    out.header = header_to_ros(bus.header)
    out.wrench = wrench_to_ros(bus.wrench)
    return out


class GeometryMsgsWrenchStampedMapper:
    _ros_type = None

    def ros_msg_type(self):
        cls = type(self)
        if cls._ros_type is None:
            from geometry_msgs.msg import WrenchStamped as RosMsg

            cls._ros_type = RosMsg
        return cls._ros_type

    def ros_to_bus(self, msg) -> bytes:
        return wrench_stamped_to_bus(msg).SerializeToString()

    def bus_to_ros(self, payload: bytes):
        BusMsg = _bus_cls()
        bus = BusMsg()
        bus.ParseFromString(payload)
        return wrench_stamped_to_ros(bus)
